services/repository/src/main/java/org/sagebionetworks/repo/web/service/table/TableServicesImpl.java

```diff
@@ -169,10 +171,17 @@ public TableFileHandleResults getFileHandles(Long userId, RowReferenceSet fileHa
 	public String getFileRedirectURL(Long userId, String tableId, RowReference rowRef, String columnId) throws IOException, NotFoundException {
 		Validate.required(columnId, "columnId");
 		Validate.required(userId, "userId");
+		
+		UserInfo userInfo = userManager.getUserInfo(userId);
 
-		String fileHandleId = getFileHandleId(userId, tableId, rowRef, columnId);
-		// Use the FileHandle ID to get the URL
-		return fileHandleManager.getRedirectURLForFileHandle(fileHandleId);
+		String fileHandleId = getFileHandleId(userInfo, tableId, rowRef, columnId);
+		
+		FileHandleUrlRequest urlRequest = new FileHandleUrlRequest(userInfo, fileHandleId)
+				.withAssociation(FileHandleAssociateType.TableEntity, tableId)
+				// Is the check performed in the AuthorizationManager for ENTITY correct?
+				.withBypassAuthCheck(true);

```

Comment:
Run the check.  For PLFM-5240, we do not want anonymous users to download data from tables even if a table is marked as "open access".  This would be plugging a potential security hole.